server/modules/surveyRdb/persistence/dbActions/nodesInsert.js

```diff
@@ -9,19 +10,48 @@ const SchemaRdb = require('../../../../../common/surveyRdb/schemaRdb')
 
 const DataTable = require('../schemaRdb/dataTable')
 
-const run = async (survey, nodeDef, record, client) => {
-  const nodeDefParent = Survey.getNodeDefParent(nodeDef)(survey)
+const flatten = array => {
```

Comment:
you can use R.flatten

Reply:
R.flatten "flattens" too much... e.g. [[[1,2,3],[4,5,6]],  [7,8,9],[10,11,12]] becomes [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], while we need to have [ [1,2,3],[4,5,6],[7,8,9],[10,11,12]] instead

Reply:
ok!!